pymongo/monitoring.py

```diff
@@ -1494,6 +1523,7 @@ def publish_command_start(
         connection_id: _Address,
         op_id: Optional[int] = None,
         service_id: Optional[ObjectId] = None,
+        server_connection_id: Optional[int] = None,
```

Comment:
If the server connection id is always passed here then perhaps we should make it a required argument? Same for the succeeded/failed methods.

Reply:
`server_connection_id` was added in MongoDB 4.2. Do we still need to support versions older than that? If so, we need to keep this optional.

Reply:
I meant that we are still always passing it to these methods even when server_connection_id is None. So we should consider removing the default value since the default is never used. Now that I'm looking at it the same is true for service_id and database_name so I'll open a new issue for this.